platform/core.windows/src/org/netbeans/core/windows/services/NbPresenter.java

```diff
@@ -15,6 +15,9 @@
  * KIND, either express or implied.  See the License for the
  * specific language governing permissions and limitations
  * under the License.
+ *
+ * Contains a bug fix contributed by Experian. © Experian, 2019. Licensed under Apache 2.0.
```

Comment:
I don't think that we should add this here.

@matthiasblaesing What do you think?

Reply:
Build failed because of that but I've asked to add it to comply the license. Happy to remove it if it's not needed.

**Request I received**

Notice changes needed
Clauses 4(b) and (c) of the licence are the relevant clauses. 4(b) says:
 
_“You must cause any modified files to carry prominent notices stating that You changed the files”_
 
How we comply with this depends how the file is structured. If the notice for a piece of code is found at the start of that particular code, then it can be a simple notice:
 
_Contains a bug fix contributed by Experian. © Experian, 2019. Licensed under Apache 2.0._
 
If all the notices are together at the front of the file, then our notice will also need to describe the particular code we’ve fixed. This notice format would work:
 
_NAME OF THE PARTICULAR CODE contains a bug fix contributed by Experian. © Experian, 2019. Licensed under Apache 2.0._
 
4(c) says that we must retain all notices already present in the software. Hopefully more straightforward! Please let me know if you would like to discuss.

Reply:
I'm not an expert about legal problems, sorry. So, I could be wrong. The license header would be an irregular pattern. If we have to change the license header in the future, we have to consider irregular patterns. It is the reason why I wrote "here". 

I think maybe Matthias(@matthiasblaesing ) could advise something. Let's wait :)

BTW, probably, that is not the cause of build failure because other PRs also fail.

Reply:
Yes. This causes the build to fail. Is that special license notice really necessary, @YannLeCorse?

Reply:
To my understanding: This code is owned by the ASF. The ASF either requires contributions to be to small to create a separate copyright (I'm not sure that I like the term "trivial" for bugfixes) or to be explicitly donated to the ASF. So if you don't add the above line and propose a PR, I would consider it donated to the ASF. It is also small enough to be called a trivial bugfix (it is work, it is appreciated, but it is small). This of course assumes, that you are in a position to donate the code, if you are not, please indicate so.
TL;DR: For code we don't import from a third party (and thus don't own), ensure that you are the author, you donate the code to the ASF, so please stay with the normal ASF headers.

Reply:
It's a stupid bug fix (trivial is too kindly), of course it's not necessary but I've been asked to evaluate our FOSS contribution process. I'll check again with my legal to get rid of it. 

Any comments for the rest ?